debugtools/DDR_VM/src/com/ibm/j9ddr/AuxFieldInfo29.dat

```diff
@@ -90,6 +90,7 @@ J9Class.instanceLeafDescription = required
 J9Class.lockOffset = required
 J9Class.module = required
 J9Class.nextClassInSegment = required
+J9Class.nullRestrictedArrayClass = required
```

Comment:
This field cannot be `required` (it's only present when `defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)`). I expect we want:
```
J9Class.nullRestrictedArrayClass = J9Class*
```
The DDR code which uses this field must anticipate `NoSuchFieldException`.

Reply:
I added a new method ValueTypeHelper.areFlattenableValueTypesSupported to use around new fields so NoSuchFieldException shouldn't happen. I'm wrapping a potential NoSuchFieldException error in CorruptDataException since if that were to happen now something is pretty wrong.